games/games.py

```diff
@@ -0,0 +1,323 @@
+    def resource_string(self, path):
+        """Handy helper for getting resources from our kit."""
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode("utf8")
+
+    def student_view(self, context=None):
+        """
+        The primary view of the GamesXBlock, shown to students
+        when viewing courses.
+        """
+        html = self.resource_string("static/html/games.html")
+        frag = Fragment(html.format(self=self))
+        frag.add_css(self.resource_string("static/css/games.css"))
+        frag.add_javascript(self.resource_string("static/js/src/games.js"))
+        frag.initialize_js('GamesXBlock')
+        return frag
+    
+    #Universal handlers--------------------------------------------------------------
+    @XBlock.json_handler
+    def expand_game(self, data, suffix=''):
+        """
+        A handler to expand the game from its title block.
+        """
+        description = "ERR: self.type not defined or invalid"
+        if self.type == "flashcards":
+            description = "Click each card to reveal the definition"
+        elif self.type == "matching":
+            description = "Match each term with the correct definition"
+        return {
+            'title': self.title,
+            'description': description,
+            'type': self.type
+        }
+    
+    @XBlock.json_handler
+    def start_game(self, data, suffix=''):
+        """
+        A handler to begin the game.
+        """
+        return {
+            'term_image': self.list[self.list_index]['term_image'],
+            'definition_image': self.list[self.list_index]['definition_image'],
+            'term': self.list[self.list_index]['term'],
+            'list_length': self.list_length,
+        }
+
+    @XBlock.json_handler
+    def close_game(self, data, suffix=''):
+        """
+        A handler to close the game to its title block.
+        """
+        if self.type == "flashcards":
+            self.term_is_visible=True
+            self.list_index=0
+        return {
+            'title': self.title
+        }
+    
+    @XBlock.json_handler
+    def display_help(self, data, suffix=''):
+        """
+        A handler to display a tooltip message above the help icon.
+        """
+        message = "ERR: self.type not defined or invalid"
+        if self.type == "flashcards":
+            message = "Click each card to reveal the definition"
+        elif self.type == "matching":
+            message = "Match each term with the correct definition"
+        return {'message': message}
+
+    
+    #Flashcards handlers--------------------------------------------------------------    
+    @XBlock.json_handler
+    def flip_flashcard(self, data, suffix=''):
+        """
+        A handler to flip the flashcard from term to definition
+        and vice versa.
+        """
+
+        #flip term_is_visible first to show definition
+        self.term_is_visible = not(self.term_is_visible)
```

Comment:
```suggestion
        if self.term_is_visible:
            self.term_is_visible = not(self.term_is_visible)
            return {'image': self.list[self.list_index]['definition_image'], 'text': self.list[self.list_index]['definition']}
       else:
            self.term_is_visible = not(self.term_is_visible)
            return {'image': self.list[self.list_index]['term_image'], 'text': self.list[self.list_index]['term']}
```

This will keep the logic a little cleaner and more readable